src/main/java/org/apache/commons/cli/DefaultParser.java

```diff
@@ -228,6 +228,9 @@ private DefaultParser(final boolean allowPartialMatching,
      */
     private void checkRequiredArgs() throws ParseException {
         if (currentOption != null && currentOption.requiresArg()) {
+            if (isJavaProperty(currentOption.getKey()) && currentOption.getValues().length == 1) {
```

Comment:
Hi @Claudenw 
I think you are missing a test case to account for `Option#getValues()` returning `null`.